### ISSUE TITLE
[Question] Do we have any new info or ETA on ingame integrated Multiplayer

### ISSUE THREAD

**Issue by matejkar** [tsunamayo/Starship-EVO#4362]
Title.
First off, if i missed something in the recent QnA the dev did, i apologise.
Since launch the multiplayer in the game is not really functioning at all without the use of ZeroTier, and even with that, its very inconsistent.
So i wish to ask for some updates, info, ETA or anything else about the current state of multiplayer and if its being worked on. If multiplayer is not worked on, can we get a rough idea on when improvements could be made, and when it could become a fully functioning part of the game.

**Comment by ProPeach** [comment]
Here's the question on multiplayer from the Q&A, there are questions and timestamps in the comments but I've timestamped this link for you
https://youtu.be/tdjH_gyQnjc?t=2231

**Comment by matejkar** [comment]
> Here's the question on multiplayer from the Q&A, there are questions and timestamps in the comments but I've timestamped this link for you https://youtu.be/tdjH_gyQnjc?t=2231

Ah i see, tnx. I would still like @tsunamayo 's word on this if its not a problem

**Comment by tsunamayo** [comment]
@matejkar I will try to do a bugfix pass soon. What issue you guys are you currently having? You cant connect?

**Comment by matejkar** [comment]
@tsunamayo thanks for the heads up. We haven't tried  multi for a few months at all now, but the ingame multiplayer all by itself never worked for us. As i said ZeroTier is currently only option. And the last time we tried on ZeroTier we had desyncs, disconnects and crashes. And i honestly think that multiplayer would boost the game's poplularity by a lot, as people would be playing it and buying it for multiplayer with friends more.

TLDR, Havent tried multi in a while, and the last time we did i did not work properly at all. Multiplayer without ZeroTier is not possible at all

**Comment by ProPeach** [comment]
I've played as a client quite a few times, although not recently. We were able to connect without using 3rd party tools like ZeroTier/Hamachi

The main issues I remember are not being able to spawn in new blueprints without causing a desync, and not being able to see any ships being piloted by other players - The ship being piloted would just stay stationary for all players apart from the one piloting it, until they stop piloting it.
I'll make it known that you're looking for bug reports and see if anyone can give log evidence to help

**Comment by wagskydive** [comment]
@ProPeach I had a very similar experience about 3 months ago playing with a friend in multiplayer 
We also couldn't really play because of these issues
@tsunamayo I will try to reproduce this and report if i find the time

**Comment by Kaiser-Indrasil** [comment]
I tried to play MP a few times and every time there was a major issue of desync.
I played as a host and my guests couldn't see either me or my ships, despite us being in the same place.
Or, they could see the ships, but they couldn't see them move.
Or, they could see the ships move, but they couldn't see them fire their weapons.

**Comment by Kaiser-Indrasil** [comment]
@tsunamayo I tried to play MP a few times and every time there was a major issue of desync.
I played as a host and my guests couldn't see either me or my ships, despite us being in the same place.
Or, they could see the ships, but they couldn't see them move.
Or, they could see the ships move, but they couldn't see them fire their weapons.

**Comment by Vicomt** [comment]
@tsunamayo Oojimaflip and I are also waiting on ANY kind of MP implementation that works. I personally won't use a separate VPN service like ZeroTier, and the attempts we've tried to join games have all been abject failures. the client attempts to connect, and the host notices a very brief partial display of the client's character, but then the client immediately disconnects and returns to the main menu.

**Comment by tsunamayo** [comment]
@matejkar @Vicomt @wagskydive @Kaiser-Indrasil @ProPeach Hi, so I have finally made some fixes for 22w04a. I think we will still find a lot of issues but I was able to connect, build, see npcs ship moving, fly and spawn ships. Please book any issue you might find!
Thanks
